docs/usage/loading-packages.md

```diff
@@ -5,81 +5,62 @@
+- {any}`pyodide.loadPackage` (Javascript) for packages built with Pyodide. This
+  is a function with less overhead but also more limited functionality.
+  micropip uses this function to load Pyodide packages. In most cases you
+  should be using micropip.
+
+In some cases, and in particular in the REPL, packages are installed implicitly
+from imports. The Pyodide REPL uses {any}`pyodide.loadPackagesFromImports` to
+automatically download all packages that the code snippet imports. This is
+useful since users might import unexpected packages in REPL. At present,
 {any}`loadPackagesFromImports <pyodide.loadPackagesFromImports>` will not
 download packages from PyPI, it will only download packages included in the
 Pyodide distribution. See {ref}`packages-in-pyodide` to check the full list of
 packages included in Pyodide.
 
-## Loading packages with {any}`pyodide.loadPackage`
+## How to chose between `micropip.install` and `pyodide.loadPackage`?
 
-Packages included in the official Pyodide repository can be loaded using
-{any}`pyodide.loadPackage`:
+While {any}`micropip.install` is written in Python and
+{any}`pyodide.loadPackage` in Javascript this has no incidence on when
+to use each of these functions. Indeed, you can easily switch languages using
+the {ref}`type-translations` with,
 
-```js
-await pyodide.loadPackage("numpy");
-```
+- from Javascript,
+  ```javascript
+  let micropip = pyodide.pyimport(package_name);
+  ```
+- from Python,
+  ```
+  import pyodide_js
+  await pyodide_js.loadPackage('package_name')
+  ```
 
-It is also possible to load packages from custom URLs:
+Instead, the general advice is to use {any}`micropip.install` for everything
+except in the following cases where {any}`pyodide.loadPackage` might be more
+appropriate,
 
-```js
-await pyodide.loadPackage(
-  "https://foo/bar/numpy-1.22.3-cp310-cp310-emscripten_3_1_13_wasm32.whl"
-);
-```
-
-The file name must be a valid wheel name.
-
-When you request a package from the official repository, all of the package's
-dependencies are also loaded. Dependency resolution is not yet implemented when
-loading packages from custom URLs.
-
-In general, loading a package twice is not permitted. However, one can override
-a dependency by loading a custom URL with the same package name before loading
-the dependent package.
-
-Multiple packages can also be loaded at the same time by passing a list to
-{any}`pyodide.loadPackage`.
-
-```js
-await pyodide.loadPackage(["cycler", "pytz"]);
-```
-
-{any}`pyodide.loadPackage` returns a `Promise` which resolves when all the
-packages are finished loading:
-
-```javascript
-let pyodide;
-async function main() {
-  pyodide = await loadPyodide();
-  await pyodide.loadPackage("matplotlib");
-  // matplotlib is now available
-}
-main();
-```
-
-(micropip)=
+- to load micropip itself,
+- when you are optimizing for size, do not want to install the `micropip`
+  package, and do not need to install packages from PyPI.
```

Comment:
This may not need to be said here, but you can use `loadPackage` to load wheels directly from PyPI as long as you also load their dependencies. Particularly when there are deep dependency trees this can be a lot faster because it requires 2n sequential requests to load a tree of depth n and particularly if early packages are big, downloading all of the wheels sequentially could take a while.

Reply:
Yes, good point. But similarly, you could pass this list of URLs to micropip with `deps=False` and it would be faster. And it also makes for a simpler overall message. I adjusted this sentence a bit.

Reply:
I leave this out for now, because this is basically the use case of loading freezed packages, and we need to improve the explanations of how to do that anyway.